src/pipx/package_specifier.py

```diff
@@ -161,7 +161,24 @@ def parse_specifier_for_install(package_spec: str, pip_args: List[str]) -> Tuple
         )
         pip_args.remove("--editable")
 
-    return (package_or_url, pip_args)
+    pip_args = [option for pip_arg in pip_args for option in pip_arg.split(" ")]
```

Comment:
This doesn't seem necessary.

```suggestion
```

Reply:
If you have more than 1 pip args, the function get it as a single string.

https://pipx.pypa.io/stable/examples/ here for example there is the example:
`pipx install --pip-args='--index-url=<private-repo-host>:<private-repo-port> --trusted-host=<private-repo-host>:<private-repo-port>' private-repo-package`

In this example the `pip_args` argument will be:
```python
['--index-url=<private-repo-host>:<private-repo-port> --trusted-host=<private-repo-host>:<private-repo-port>']
```

So from this I assumed I need first to check for multiple flags in a single string and split it :)

Reply:
> In this example the `pip_args` argument will be:
> 
> ```python
> ['--index-url=<private-repo-host>:<private-repo-port> --trusted-host=<private-repo-host>:<private-repo-port>']
> ```

That doesn't seem correct. The `pip` arguments get normalised and formatted already at an early stage:

https://github.com/pypa/pipx/blob/bd69963a4dbd90c95950921e99654dc9867ef10f/src/pipx/main.py#L161

Reply:
ok now I see, so it another bug... `shlex.split()` does not split anything between single quotes on posix platforms.

on shlex documentation:

> Enclosing characters in quotes which are not part of [escapedquotes](https://docs.python.org/3/library/shlex.html#shlex.shlex.escapedquotes) (e.g. "'") preserve the literal value of all characters within the quotes;

Do you have any idea for appropriate way to resolve it?

Since I dont know what exactly happening on windows I dont want just to strip the single quotes (or changing to regular function `string.split()`)

Reply:
also, what about the `-c` flag? (or other single dash flags)

now I understand I did wrong on the `-c` as on this flag should not be  with`=`... (e.g `-c some_file.txt`)

Reply:
> ok now I see, so it another bug... `shlex.split()` does not split anything between single quotes on posix platforms.

The passage you quoted means that any special characters in the string are preserved. It does not result in an incorrect splitting behaviour. I can assure you that works, as I reproduced it locally.

Reply:
Are you saying you running this line and it split the strings to you?

```python
shlex.split("'-c .github/workflows/constraints.txt'")
```

Please double check it...

If so, do you have any idea why it does not work for me? maybe it related to python version? (using python 3.12.2)

Reply:
The single quotes given on the command line aren't preserved when given to `shlex`, or rather there is no difference to double quotes in this case – it just tells the program (the argument parser) that the arguments given  belong as one single string to the option.

Reply:
Im sorry, but Im not sure I understand what do you mean...

I do the following steps:

1. install `pipx` locally
2. put print for pip_args inside `parse_specifier_for_install` function
3. calling `pipx install --pip-args='-c a' foo`
4. see printing this `['-c a']`

I do something wrong?
maybe we could talk at discord or something?

Reply:
I debugged the `pip_args` in `parse_specifier_for_install` directly, too (POSIX mode), calling exactly the same command, but it worked for me. What OS are you running? I can also investigate with `pyenv`, but unsure whether it's related to the Python version.

Reply:
I just realized it's probably because im using `xonsh` shell.

check it on `zsh` and it split the `pip_args` correctly.

I don't fully understand why this is happening on xonsh, but I will try to figure it out or maybe talk with them.

anyway is seems it does not relate to `pipx` or to this PR and I think we can proceed :)